### PR TITLE
add tx_formlog_entries to garbage collect task

### LINKED ISSUE #11: Add tx_formlog_entries to TableGarbageCollectionTask
Related to #5 

It should be easily possible to automatically remove old entries from fromlog.
The most out-of-the-box way I know of would be to add it to the `TableGarbageCollectionTask`.

I'm thinking of something like this:

```PHP
$GLOBALS['TYPO3_CONF_VARS']['SC_OPTIONS']['scheduler']['tasks'][\TYPO3\CMS\Scheduler\Task\TableGarbageCollectionTask::class]['options']['tables']['tx_formlog_entries'] = [
    'dateField' => 'tstamp',
    'expirePeriod' => 90,
];
```

I'm not sure where I read it but 90 days was how long you should keep the data but also when you should delete it.

### PR DESCRIPTION
**Pull request by Nemo64**
resolves #11 

However there is a pitfall: if someone has the TableGarbageCollectTask configured with "allTables", then this will potentially remove a lot of data without him knowing.

Also we can debate about the 90 days. Typo3's default value for sys_log is 180 days.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -37,6 +37,9 @@
     "formlog": "self.version",
     "typo3-ter/formlog": "self.version"
   },
+  "suggest": {
+    "typo3/cms-scheduler": "Allows automatic garbage collection of old form entries."
```
Comment (mbrodala) [inline composer.json]: Can you make this _... form log entries._ please?

File: ext_localconf.php [hunk 2]
```diff
@@ -9,3 +9,8 @@
 
 $GLOBALS['TYPO3_CONF_VARS']['SYS']['locallangXMLOverride']['EXT:form/Resources/Private/Language/Database.xlf'][1519643592] = 'EXT:formlog/Resources/Private/Language/Database.xlf';
 $GLOBALS['TYPO3_CONF_VARS']['SYS']['locallangXMLOverride']['de']['EXT:form/Resources/Private/Language/Database.xlf'][1519643592] = 'EXT:formlog/Resources/Private/Language/de.Database.xlf';
+
+$GLOBALS['TYPO3_CONF_VARS']['SC_OPTIONS']['scheduler']['tasks']['TYPO3\\CMS\\Scheduler\\Task\\TableGarbageCollectionTask']['options']['tables']['tx_formlog_entries'] = [
```
Comment (mbrodala) [inline ext_localconf.php]: I'd still prefer the `::class` notation and will adjust the static analysis configuration accordingly, please change this back afterwards.

**Comment by Nemo64** [comment]
You basically did your requested changes yourself so...

**Comment by mbrodala** [comment]
Yeah, I had some time to spend. Do you have any source for the 90 days wrt GDPR? Since it's a default most people will use as is it should be sane.

**Comment by Nemo64** [comment]
@mbrodala nope, i don't. seems like it was just internal half-knowledge. So I'd suggest using the typo3 sys_log default of 180 here too.

**Comment by mbrodala** [comment]
Thanks a lot. :-)

**Pull request merged**